internal/serve/httphandler/verifiy_receiver_registration_handler.go

```diff
@@ -112,6 +112,25 @@ func (v VerifyReceiverRegistrationHandler) processReceiverVerificationPII(
 	now := time.Now()
 	truncatedPhoneNumber := utils.TruncateString(receiver.PhoneNumber, 3)
 
+	// STEP 0: if customer-id exists in sep24 claims, use it to look up reciever, hash Phone Number  in db, compare against hashed mobile_number in sep24 claims
+	if receiverRegistrationRequest.CustomerID != "" {
+		receiverByExternalID, err := v.Models.Receiver.GetByExternalID(ctx, dbTx, receiver.ExternalID)
+		if err != nil {
+			log.Ctx(ctx).Errorf("Error retrieving receiver by external ID: %v", err)
+			return &ErrorInformationNotFound{cause: err}
+		}
+		fmt.Println(receiverByExternalID.PhoneNumber)
+
+		if data.CompareVerificationValue(receiverRegistrationRequest.MobileNumberHash, receiverByExternalID.PhoneNumber) {
+			// Compare the hashed phone number with the hashed mobile number from the SEP-24 claims
+			//if hashedPhoneNumber != receiverRegistrationRequest.MobileNumberHash {
+			//err := fmt.Errorf("hashed phone number does not match the hashed mobile number from SEP-24 claims")
+			//	return &ErrorInformationNotFound{cause: err}
+			//}
+			return nil
+		}
```

Comment:
👋 👋 👋  This logic is incorrect. When you return nil here, you're skipping all the check from the following steps (STEP 1 -> 4).

What you should be doing here is returning an error if the comparison fails, similar to what's in the comment:
```suggestion
		if !data.CompareVerificationValue(receiverRegistrationRequest.MobileNumberHash, receiverByExternalID.PhoneNumber) {
			err := fmt.Errorf("hashed phone number does not match the hashed mobile number from SEP-24 claims")
			return &ErrorInformationNotFound{cause: err}
		}
```